### ISSUE TITLE
[RPI] add hw decoding support for VC-1

### ISSUE THREAD

**Issue by tru** [plexinc/plex-media-player#16]
https://forums.plex.tv/discussion/188472/vc1-raspberry-pi2#latest

**Comment by wm4** [comment]
I don't expect it to be hard, but requires a paid license key.

**Comment by alxscott** [comment]
Just to reiterate what I posted in the forum above, and to push this more in the "Bug" direction than feature request...

The Pi gets away with such a low power chip through allowing hardware decoding of most common codecs, VC-1 being one of them. I think hardware decoding ( with purchased license as a given) is a definite requirement if RPi2 is going to be a fully supported client platform. If VC-1 isn't supported on Raspberry Pi it may severely limit its usage... I know a fair portion of my older full rip Blu Rays are VC-1 and I think reflects the wider scope of Blurays out here. 

Thanks for all your efforts... Any testing required let me know!

**Comment by LongChair** [comment]
@tru, @wm4 : what we might need to do is to detect if RPI supports VC1 & MPEG2. there are some shell commands for this. I could take care of this. Then the other part is how do we give this information to the transcoding profile ...

**Comment by wm4** [comment]
There are some mysteries vc commands, which can query the license key status. Kodi/XBMC uses this.

@dogonwheels: this means web-client has to call into the C++ host, and dynamically adjust the transcode profile on start. Based on what the C++ host answers, you have to add either VC-1 or mpeg2, or both. Is this possible?

**Comment by LongChair** [comment]
@vlang : yes i already used those in PHT for rasplex. Just not sure how to interface that with web. We'd need to have some "capabilities" interface with web so that we can give some info for MDE.

**Comment by dogonwheels** [comment]
Can we pass (via channel, user agent or otherwise) the license details to web on startup? Would fit better than an async call on playback for capabilities.

**Comment by wm4** [comment]
Would a function that returns a list of enabled codecs work?

**Comment by dogonwheels** [comment]
_Maybe_ - I'll take a look later - but that will still require a channel call, which I don't think the profiles (yet) support. We'll work it in somehow!

**Comment by wm4** [comment]
@dogonwheels: OE build #178 should now have this API: https://github.com/plexinc/plex-media-player/blob/cf6aa280b90c5b760b7482be6688165264520ae1/src/player/PlayerComponent.h#L93

The codec names should match with the names we use everywhere. I refrained from keeping a complete list of supported codecs; it wouldn't be too useful, and this API actually just signals whether "fast" decoding for these specific 2 codecs (vc1, mpeg2) is available. On non-RPI platforms the new function always returns `true`.

**Comment by tru** [comment]
We should most likely bounce this over to @firasdib

**Comment by wm4** [comment]
PS: you also can add `mpeg4` to the codecs that can always be decoded. I'm not sure if the RPI can really decode all mpeg4 videos out there, but we can try for now.

**Comment by camjac251** [comment]
Why was this closed?

**Comment by firasdib** [comment]
This is implemented and should be working @camjac251. If it's not, report it in the forums and someone will take a look at it.

**Comment by camjac251** [comment]
How would I go about using it? For me, VC1 support in PMP with the Raspberry Pi2 on 1.1.1 didn't work. I had used the image from the official Plex site. VC1 video on directplay would start playing then freeze after a few seconds (player was fine, the video and audio froze like it paused). I then tried Rasplex with the vc1 license configured and it plays just fine.

Was it not available on that server? I had also noticed a few issues with AVC/DTSHDMA videos as well on directplay. I'm guessing the Pi2 isn't able to keep up
